### PR TITLE
1108 update references to admiraltest data

### LINKED ISSUE #1088: Suggestion: Rename {admiral} packaged ADaM Dataset
You've package ADaM data within {admiral}, which is great - but I have some concerns about keeping standard ADaM dataset names for these dataset within the package. For example - this is the issue that I've just run into:

- Reviewing code from someone else
- Dataset named advs_temp is created and read from external source
- Subsequent code references advs, not advs_temp
- All code runs without error

So, this would be a potentially common scenario within a clinical workflow. A user might try to reference advs directly, or any other adam dataset named within {admiral}, and the data will be found. But for novice users - or even experienced users like myself that weren't considering the ADaM data in admiral - this can cause some confusion. With the stack of the namespaces, the data are found and there's no inherent message that the data being referenced are pulled from {admiral}. The object is just found.

So my suggestion is to alter the names of the data so that they're unique and obviously belong within {admiral}. While packaging data is common within other pages (i.e. ggplot and the mpg dataset), within our industry specific workflows, it's pretty likely that you'll be working with datasets by these names exactly.

### LINKED ISSUE #1108: Update all references to {admiraltest} data with prefix `admiral_`.
As discussed in this [issue](https://github.com/pharmaverse/admiral/issues/1088) and subsequently in this [issue and associated PR](https://github.com/pharmaverse/admiraltest/issues/30):

- Data in {admiraltest} has been updated to `admiral_xx` and therefore any reference to data from {admiraltest} in vignettes, example scripts, etc. will need to be updated accordingly.
For example, `data('ae')` will need to become `data('admiral_ae')`.

- ADaM Datasets packaged as part of {admiral} need to be prefixed with `admiral_`, e.g. `adsl` becomes `admiral_adsl`

- As a result, any scripts that reference (or create) these programs need to be updated also.

### PR DESCRIPTION
**Pull request by bms63**
Closes #1108 Closes #1088

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi all,

This PR changes a lot of files is small ways.  Hoping we can go over with a fine-tooth comb.

I'm still a bit flummoxed with the `create_query_data.R`  error.  If anyone has suggestions or can pull down and see if it works in there environment, then please! :)  
image

File: R/derive_var_extreme_date.R [hunk 1]
```diff
@@ -109,44 +109,45 @@ derive_var_lstalvdt <- function(dataset,
...
-#' data("adsl")
+#' data("admiral_dm")
+#' data("admiral_ae")
+#' data("admiral_lb")
+#' data("admiral_adsl")
 #'
 #' # derive last known alive datetime (LSTALVDTM)
 #' ae_start <- date_source(
-#'   dataset_name = "ae",
+#'   dataset_name = "admiral_ae",
```
Comment (thomas-neitmann) [inline R/derive_var_extreme_date.R:119]: The `dataset_name` should still just be `"ae"`. However, the list you pass as the `source_datasets` argument below should be `list(ae = admiral_ae)`.

Comment (thomas-neitmann) [reply R/derive_var_extreme_date.R:119]: Same for all other dataset names.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: inst/templates/ad_adae.R [hunk 2]
```diff
@@ -14,11 +14,15 @@ library(lubridate)
...
+data("admiral_adsl")
 data("ex_single")
 
+adsl <- admiral_adsl
+ae <- admiral_ae
+suppae <- admiral_suppae
+
 ae <- convert_blanks_to_na(ae)
 suppae <- convert_blanks_to_na(suppae)
 ex <- convert_blanks_to_na(ex_single)
```
[the remark this answers is not in the document]
Comment (bms63) [reply inst/templates/ad_adae.R:28]: We were thinking that someone might grab that code for their work. So the person wouldn't have to change `admiral_ae `to `ae `in their code.

File: R/derive_var_extreme_date.R [hunk 3]
```diff
@@ -190,11 +191,12 @@ derive_var_lstalvdt <- function(dataset,
...
 #' )
 #'
-#' dm %>%
+#' admiral_dm %>%
 #'   derive_var_extreme_dtm(
 #'     new_var = LSTALVDTM,
 #'     ae_start, ae_end, lb_date, adsl_date,
-#'     source_datasets = list(adsl = adsl, ae = ae, lb = lb),
+#'     source_datasets = list(admiral_adsl = admiral_adsl,
+#'     admiral_ae = admiral_ae, admiral_lb = admiral_lb),
```
[the remark this answers is not in the document]
Comment (thomas-neitmann) [reply R/derive_var_extreme_date.R:199]: Exactly!

File: inst/templates/ad_adae.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomas-neitmann) [reply inst/templates/ad_adae.R:28]: Alright, sounds good 👍

**Comment by thomas-neitmann** [comment]
> Hi all,
> 
> This PR changes a lot of files is small ways. Hoping we can go over with a fine-tooth comb.
> 
> I'm still a bit flummoxed with the `create_query_data.R` error. If anyone has suggestions or can pull down and see if it works in there environment, then please! :) image

That issues seems to have been resolved after Stefan's most recent changes on {admiraltest}. Getting this new error, through. Will fix it now. 
image

**Comment by bms63** [comment]
@thomas-neitmann Thanks Thomas.  Let me know when you have fixed it.  I made some of the changes you requested and will push up once your change is done.

**Comment by bms63** [comment]
@thomas-neitmann `devtools::check()` also runs for me with no issues.

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**